Submissions & Improvement Voting Implemented

EmmaPuls opened the pull request:
I also added logic that lets people vote on old versions of the submission/ improvements by checking if the likes==null and setting likes to 0.

EmmaPuls committed:
Multiple Changes to Improvements tactics
- Improvement's form now loads on a separate page with submissionID given
- Improvements model modified to save a single submissionID rather than a list of submissionIDs
- Submissions client routes now routes to createImprovement page that contains submissionID
- Submissions view just shows a button to take you to createImprovement form
- Submissions server routes tries to route to improvements functions in the improvement server controller, but app.route appears to be using the 'api/submissions/:submissionID' route instead of the new route.
- core.css (textarea boxes now expand when in focus)

EmmaPuls committed:
Multiple Changes to Improvements tactics
- Improvement's form now loads on a separate page with submissionID given
- Improvements model modified to save a single submissionID rather than a list of submissionIDs
- Submissions client routes now routes to createImprovement page that contains submissionID
- Submissions view just shows a button to take you to createImprovement form
- Submissions server routes tries to route to improvements functions in the improvement server controller, but app.route appears to be using the 'api/submissions/:submissionID' route instead of the new route.
- core.css (textarea boxes now expand when in focus)

EmmaPuls committed:
Improvement save to db now working
- New Controller, improvement-from-submission, handles having an extra piece of data for the creation of the improvements.
- Old controller may need to be reviewed, no improvement can be created without a submissionID
- The submission and improvement views are modified so that they wrap better when on a small screen.
- Improvements.server.controller has logic for saving submissionID to a new improvement
- Improvement model is updated to include a submissionID
- Removed testing logic from the submissions area
- Submission client routes now uses the new improvement controller

EmmaPuls committed:
Improvement load list per submission on new page
- Removed uneccessary code from improvement-from-submission.client.controller
- Fixed indentation on list-improvements.client.controller & improvement-from-submission.client.controller
- Cosmetic changes to list-improvements.client.view
- improvements.server.controller get's the improvement list from the submissionID in the page URL (therefore doesn't work if you refresh the page)
- submissions.client.routes loads a new page for the improvements list
- view-submission.client.view give a link to the list of improvements for that submission

EmmaPuls committed:
Merge remote-tracking branch 'github/master'

# Conflicts:
#	modules/submissions/client/controllers/list-submissions.client.controller.js

EmmaPuls committed:
Merge branch 'master' into Improvements

EmmaPuls committed:
Improvement load list per submission on new page
- Included improvement list searching

EmmaPuls committed:
Improvement load list per submission on new page
- Included improvement list searching

EmmaPuls committed:
Improvement load list per submission on same page
- included more padding for header in css
- removed uneeded $state from ImprovementsListController
- changed improvements header to a button that loads the list of improvements for the page
- changed the submissions.view state in submissions.client.route to load the improvements list as well,
   updated the submission view to reflect this change

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

CSS:
- likes font size set to 150%
- glyphicon upvote and downvote icons resized

Client Controller:
- added functions to increment/ decrement the # of likes and update the db

List View:
- included upvote/ downvote buttons and linked them to the controller functionality
- segemented the improvement list rows so that they wrap properly on smaller screens

Server Controller:
- when an improvement is created the # of likes is initialised to 0

Server Routes:
- allows improvements to be updated from the list view

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

CSS:
- likes font size set to 150%
- glyphicon upvote and downvote icons resized

Client Controller:
- added functions to increment/ decrement the # of likes and update the db

List View:
- included upvote/ downvote buttons and linked them to the controller functionality
- segemented the improvement list rows so that they wrap properly on smaller screens

Server Controller:
- when an improvement is created the # of likes is initialised to 0

Server Routes:
- allows improvements to be updated from the list view

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

CSS:
- likes font size set to 150%
- glyphicon upvote and downvote icons resized

Client Controller:
- added functions to increment/ decrement the # of likes and update the db

List View:
- included upvote/ downvote buttons and linked them to the controller functionality
- segemented the improvement list rows so that they wrap properly on smaller screens

Server Controller:
- when an improvement is created the # of likes is initialised to 0

Server Routes:
- allows improvements to be updated from the list view

Commit:
Merge pull request #1 from EmmaPuls/Improvement-Votes

Improvement votes

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

Cosmetic adjustments

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

Cosmetic adjustments

EmmaPuls committed:
Improvements now have voting (1 point per vote, unlimited votes per user)

Cosmetic adjustments

Commit:
Merge pull request #2 from EmmaPuls/Improvement-Votes

Improvements now have voting

EmmaPuls committed:
Improvements voting (1 point per vote, unlimited votes per user)
- Cosmetic adjustments

EmmaPuls committed:
Improvements voting (1 point per vote, unlimited votes per user)
- Cosmetic adjustments

EmmaPuls committed:
Improvements voting (1 point per vote, unlimited votes per user)
- Cosmetic adjustments

EmmaPuls committed:
Submissions voting (1 point per vote, unlimited votes per user)
- Cosmetic adjustments

EmmaPuls committed:
Submissions voting (1 point per vote, unlimited votes per user)
- Cosmetic adjustments

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list improvements view
  - new line for date created
- list submissions controller
  - == becomes ===
- list submissions view
  - padding and margins edited for cosmetic reasons
  - fixed linking to the submissions view from list click
  - new line for date created

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list improvements view
  - new line for date created
- list submissions controller
  - == becomes ===
- list submissions view
  - padding and margins edited for cosmetic reasons
  - fixed linking to the submissions view from list click
  - new line for date created

laur33 committed:
Checking if branch commit/push is successful

laur33 committed:
round 2 of making sur ei can commit a branch

laur33 committed:
round 2 of making sur ei can commit a branch

laur33 committed:
Backend for user comments

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list improvements view
  - new line for date created
- list submissions controller
  - == becomes ===
- list submissions view
  - padding and margins edited for cosmetic reasons
  - fixed linking to the submissions view from list click
  - new line for date created

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list improvements view
  - new line for date created
- list submissions controller
  - == becomes ===
- list submissions view
  - padding and margins edited for cosmetic reasons
  - fixed linking to the submissions view from list click
  - new line for date created

jasbro1 commented:
Looks good. I have a few suggestions before it's merged to master though:
- In the list-submissions file == is used, I think this should be === (== and === are different in Javascript)
- If a user that is not logged in tries to vote then they are redirected to a 'not authorised' page, would it maybe be better to direct them to the login page or just display a popup message saying login first?
- After creating a improvement the user is redirected to a page that just has that improvement, do you think it would make more sense to redirect the user back to the page that the submission page that they submitted the improvement on?
- In the improvement it says "Posted on Jason Brown Apr 25, 2016". The name and date should be the other way around.
- The create improvement button is shown to users who aren't logged in (these users can click on the button and type out an improvement but then get told they are not authorised when trying to submit the improvement).

EmmaPuls committed:
Merge remote-tracking branch 'origin/Submissions-Votes' into Submissions-Votes

# Conflicts:
#	modules/submissions/server/routes/submissions.server.routes.js

EmmaPuls committed:
Cosmetic adjustments & logic fixes
                               - == becomes ===
                             - list submissions view
                               - now orders by popularity first

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list-improvements view
  - Posted on becomes Posted by

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list-improvements view
  - Sort by most popular

EmmaPuls committed:
Merge branch 'Submissions-Votes' into Improvement-Votes

EmmaPuls committed:
Cosmetic adjustments & logic fixes
- list-improvements view
  - Sort by popularity
- fixed indentations in usercomments module

EmmaPuls committed:
Guest access restrictions
- list-improvements controller & view
 - guests cannot see upvote and downvote buttons
 - users who are logged in can see and use upvote and downvote buttons
 - submission controller & view
  - guests cannot see the Add Improvement button

 Recommendations:
 - After creating a improvement the user is redirected to a page that just has that improvement, do you think it would make more sense to redirect the user back to the page that the submission page that they submitted the improvement on?
  - How to implement this is currently unknown

EmmaPuls commented:
Hey @jasbro1 ,

As you mentioned I have implemented all of your recommendations with the exception of:
- **After creating a improvement the user is redirected to a page that just has that improvement, do you think it would make more sense to redirect the user back to the page that the submission page that they submitted the improvement on?**

I do not know how to implement this I spend a few hours trying with no luck, if you/ anyone else wants to have a look at it I would be much obliged. 

Two further recommendations I would add is:
- Sort by popularity by default (only a problem with improvements, already working in submissions) [_optional_]
- **Link to the individual view from the list, for editing the text (again only a problem with improvements)**

EmmaPuls commented:
Closed to re-open new branch pull request
